### PR TITLE
Feature magazine

### PR DESCRIPTION
**Pull request by fvv-vasquez**
Magazine' classes.
Enum for SQL.

### CODE REVIEW

File: src/main/java/com/fvv/bookstore/enums/SqlQueryEnum.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	MAGAZINE_SELECT_ID("SELECT * FROM tb_magazine WHERE mag_id = ?");
+	
+	private String query;
+	
+	/**
+	 * Enum constructor.
+	 * 
+	 * @param query of String type.
+	 */
+	private SqlQueryEnum(final String query) {
```
Comment (cssantos0) [inline src/main/java/com/fvv/bookstore/enums/SqlQueryEnum.java]: you can remove the private here

File: src/main/java/com/fvv/bookstore/controller/MagazineController.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+public interface MagazineController {
+
+	/**
+	 * Add a magazine using DAO.
+	 * 
+	 * @param magazine of Magazine type.
+	 * @throws ControllerException when a problem in controller happens.
+	 * @throws MagazineValidationException when a field is empty.
+	 */
+	public boolean addMagazine(final Magazine magazine) throws
```
Comment (cssantos0) [inline src/main/java/com/fvv/bookstore/controller/MagazineController.java]: Remove `public` word for all interface methods. All methods in an interface are implicitly public and abstract

File: src/main/java/com/fvv/bookstore/menu/Menu.java [hunk 3]
```diff
@@ -88,6 +94,38 @@ private void createSubMenuBooks() {
 		}
 	}
 	
+	/**
+	 * Method to create and appear the magazine sub menu.
+	 */
+	private void createSubMenuMagaziness() {
```
Comment (cssantos0) [inline src/main/java/com/fvv/bookstore/menu/Menu.java]: Please add the annotation to createSubMenuMagazines, createSubMenuBooks and createMenu methods:
```
@SuppressWarnings("InfiniteLoopStatement")
```

File: src/main/java/com/fvv/bookstore/dao/MagazineDAO.java [hunk 4]
```diff
@@ -18,27 +20,42 @@
 	 * Add a magazine at the bookstore in the database.
 	 * 
 	 * @param magazine of Magazine type.
+	 * @return
```
Comment (cssantos0) [inline src/main/java/com/fvv/bookstore/dao/MagazineDAO.java]: there is no return here

File: src/main/java/com/fvv/bookstore/bean/Magazine.java [hunk 5]
```diff
@@ -175,21 +175,21 @@ public Double getPrice() {
 	}
 	
 	/**
-	 * Sets the date.
+	 * Sets the modificationDate.
 	 * 
 	 * @param date the date to set
```
Comment (cssantos0) [inline src/main/java/com/fvv/bookstore/bean/Magazine.java]: it should be modificationDate here

File: src/main/java/com/fvv/bookstore/menu/Menu.java (hunk 3 above)
Comment (cssantos0) [inline src/main/java/com/fvv/bookstore/menu/Menu.java]: there is a typo in the method name

File: src/main/java/com/fvv/bookstore/controller/MagazineControllerImpl.java [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+	/**
+	 * Validate if a field is empty.
+	 * 
+	 * @param magazine of type Magazine
+	 * @throws MagazineValidationException if a field is empty.
+	 */
+	private void validateMagazine(final Magazine magazine) throws MagazineValidationException {
+		StringBuilder sb = new StringBuilder();
+		if(magazine.getName().equals("")) {
+			sb.append("Field name cannot be empty.").append("\n");
```
Comment (cssantos0) [inline src/main/java/com/fvv/bookstore/controller/MagazineControllerImpl.java]: Create a utility class called `Constants` and then add the property:

```
private static final String LINE_SEPARATOR = System.lineSeparator();
```

Finally replace all `\n` occurrences in the  code

File: src/main/java/com/fvv/bookstore/util/DateUtil.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
+package com.fvv.bookstore.util;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * Class DateUtil to convert a String into a Date
```
Comment (cssantos0) [inline src/main/java/com/fvv/bookstore/util/DateUtil.java]: a better comment here could be

Utility class to handle date operations

File: src/main/java/com/fvv/bookstore/util/DateUtil.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+ * 
+ * @author Fatima Vasquez
+ * <p>Created on 30 de mar de 2018</p>	
+ * @version 1.0 	
+ *
+ */
+public final class DateUtil {
+
+	public static Date stringToDate(final String strDate) {
```
Comment (cssantos0) [inline src/main/java/com/fvv/bookstore/util/DateUtil.java]: Add the javadoc for the method

File: src/main/java/com/fvv/bookstore/util/DateUtil.java [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+ * <p>Created on 30 de mar de 2018</p>	
+ * @version 1.0 	
+ *
+ */
+public final class DateUtil {
+
+	public static Date stringToDate(final String strDate) {		
+		if (strDate != null) {
+			try {
+				SimpleDateFormat format = new SimpleDateFormat("MM/dd/yyyy");
```
Comment (cssantos0) [inline src/main/java/com/fvv/bookstore/util/DateUtil.java]: put the mask in a separated constant attribute

File: src/main/java/com/fvv/bookstore/util/DateUtil.java [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+ * Class DateUtil to convert a String into a Date
+ * 
+ * @author Fatima Vasquez
+ * <p>Created on 30 de mar de 2018</p>	
+ * @version 1.0 	
+ *
+ */
+public final class DateUtil {
```
Comment (cssantos0) [inline src/main/java/com/fvv/bookstore/util/DateUtil.java:15]: Add a method that gets a Date and return a String formatted in the proper mask, then update the Magazine.toString() method to call this new method, so in the list output a proper mask can be exhibited.

**Review by cssantos0 — CHANGES_REQUESTED** [review]

File: src/main/java/com/fvv/bookstore/enums/SqlQueryEnum.java (hunk 1 above)
Comment (fvv-vasquez) [reply src/main/java/com/fvv/bookstore/enums/SqlQueryEnum.java]: Ok

File: src/main/java/com/fvv/bookstore/controller/MagazineController.java (hunk 2 above)
Comment (fvv-vasquez) [reply src/main/java/com/fvv/bookstore/controller/MagazineController.java]: ok

File: src/main/java/com/fvv/bookstore/dao/MagazineDAO.java (hunk 4 above)
Comment (fvv-vasquez) [reply src/main/java/com/fvv/bookstore/dao/MagazineDAO.java]: Ok

File: src/main/java/com/fvv/bookstore/bean/Magazine.java (hunk 5 above)
Comment (fvv-vasquez) [reply src/main/java/com/fvv/bookstore/bean/Magazine.java]: Ok!

File: src/main/java/com/fvv/bookstore/menu/Menu.java (hunk 3 above)
Comment (fvv-vasquez) [reply src/main/java/com/fvv/bookstore/menu/Menu.java]: Ok

File: src/main/java/com/fvv/bookstore/controller/MagazineControllerImpl.java (hunk 6 above)
Comment (fvv-vasquez) [reply src/main/java/com/fvv/bookstore/controller/MagazineControllerImpl.java]: Ok

File: src/main/java/com/fvv/bookstore/util/DateUtil.java (hunk 7 above)
Comment (fvv-vasquez) [reply src/main/java/com/fvv/bookstore/util/DateUtil.java]: Ok

File: src/main/java/com/fvv/bookstore/util/DateUtil.java (hunk 8 above)
Comment (fvv-vasquez) [reply src/main/java/com/fvv/bookstore/util/DateUtil.java]: Ok

File: src/main/java/com/fvv/bookstore/util/DateUtil.java (hunk 9 above)
Comment (fvv-vasquez) [reply src/main/java/com/fvv/bookstore/util/DateUtil.java]: Ok

File: src/main/java/com/fvv/bookstore/util/DateUtil.java (hunk 10 above)
Comment (fvv-vasquez) [reply src/main/java/com/fvv/bookstore/util/DateUtil.java:15]: Ok

**Review by cssantos0 — APPROVED** [review]
very good work

**Pull request merged**